### PR TITLE
Stabilize RetryPolicy API for OTLP exporters

### PR DESCRIPTION
**Pull request by jack-berg**
We discussed the stability status of OTLP exporter retry in the 5/25 java SIG. I mentioned that while I still don't think we can enable retry by default in autoconfigure until this [spec issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1742) is resolved, I do think we should stabilize the API configuration of retry. 

This means moving RetryPolicy, RetryPolicyBuilder to stable package / artifact, and adding `setRetryPolicy(RetryPolicy)` methods to the OTLP exporter builder classes. That's what this PR accomplishs.

There's an interesting question about where RetryPolicy should live. It currently lives in an internal package in `opentelemetry-exporters-common`, which has no public API surface area. Rather than adding new public API surface area and being stuck with that artfiact / package, I thought a better home would be `opentelemetry-sdk-common`.

### CODE REVIEW

File: exporters/common/src/test/java/io/opentelemetry/exporter/internal/retry/RetryPolicyTest.java
Comment (mateuszrzeszutek) [inline exporters/common/src/test/java/io/opentelemetry/exporter/internal/retry/RetryPolicyTest.java:1]: Shouldn't this test be moved to `sdk-common`?

Comment (jack-berg) [reply exporters/common/src/test/java/io/opentelemetry/exporter/internal/retry/RetryPolicyTest.java:1]: Oh, yes.

File: sdk/common/src/main/java/io/opentelemetry/sdk/common/export/RetryPolicyBuilder.java [hunk 1]
```diff
@@ -3,34 +3,36 @@
...
 import static java.util.Objects.requireNonNull;
 
 import java.time.Duration;
 
-/**
- * This class is internal and is hence not for public use. Its APIs are unstable and can change at
- * any time.
- */
+/** Builder for {@link RetryPolicy}. */
 public final class RetryPolicyBuilder {
```
Comment (bogdandrutu) [inline sdk/common/src/main/java/io/opentelemetry/sdk/common/export/RetryPolicyBuilder.java:14]: Can you use https://github.com/google/auto/blob/main/value/userguide/builders-howto.md#autobuilder?

Comment (jack-berg) [reply sdk/common/src/main/java/io/opentelemetry/sdk/common/export/RetryPolicyBuilder.java]: 👍

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Pull request merged**